set.mm

```diff
@@ -514828,82 +514951,93 @@ such that every prime ideal contains a prime element (this is a
-        VAVLAXMWEWGWMSZSWNAWEXLXQAWESZWTWGXEWMAWTWGVCWEAWSWFWDABCDEFGWSJKLMNXIV
-        DVEVFXRXCWKXDWLXRXBWIWJXRBXAWDCEFGKXJLMAWEVGAECVHRQZWENVFVIVJAXDWLUIZWE
-        AXSXTNECGXDKXKVKVMVFVNVOVPWEWGWMVQVLVRVSBWHFGWLWDIWFLMWFTWHTPWLTULWDBHV
-        TWAWB $.
-    $}
-
-    ressply1.1 $e |- P = ( S |`s B ) $.
-    ${
-      $d B y $.  $d P y $.  $d U y $.  $d X y $.  $d ph y $.
-      ressply1invg.1 $e |- ( ph -> X e. B ) $.
-      $( An element of a restricted polynomial algebra has the same group
-         inverse.  (Contributed by Thierry Arnoux, 30-Jan-2025.) $)
-      ressply1invg $p |- ( ph -> ( ( invg ` U ) ` X ) = ( ( invg ` P ) ` X ) )
-        $=
-        ( cfv wceq wcel eqid vy cplusg c0g crio cminusg ressply1bas ressply1add
-        cv cbs anassrs mpidan ressply10g cmnd wss csubrg crg subrgply1 subrgrcl
-        co syl ringmnd 3syl csubg subrgsubg subg0cl cps1 cin ressply1bas2 inss2
-        wa eqsstrdi ress0g syl3anc eqtr3d adantr eqeq12d riotaeqbidva grpinvval
-        eleqtrd 3eqtr4d ) AUAUHZIGUBQZUSZGUCQZRZUABUDZWAICUBQZUSZCUCQZRZUACUIQZ
-        UDZIGUEQZQZICUEQZQZAWEWJUABWKABCDEFGHJKLMNOUFZAWABSZVJWCWHWDWIAWRIBSZWC
-        WHRZPAWRWSWTABCDEFGHWAIJKLMNOUGUJUKAWDWIRWRAEUCQZWDWIABDEFGHXAJKLMNXATZ
-        ULAEUMSZXABSZBEUIQZUNXAWIRABEUOQSZEUPSXCAFDUOQSXFNBDEFGHJKLMUQUTZBEUREV
-        AVBAXFBEVCQSXDXGBEVDBEXAXBVEVBABHVFQZUIQZXEVGXEABXIDEFGHXEXHJKLMNXHTXIT
-        XETZVHXIXEVIVKBXEECXAOXJXBVLVMVNVOVPVQAWSWNWFRPUABWBGWMIWDMWBTWDTWMTVRU
-        TAIWKSWPWLRAIBWKPWQVSUAWKWGCWOIWIWKTWGTWITWOTVRUTVT $.
-    $}
-
+    ply1asclunit.1 $e |- P = ( Poly1 ` R ) $.
+    ply1asclunit.2 $e |- A = ( algSc ` P ) $.
+    ply1asclunit.3 $e |- B = ( Base ` R ) $.
+    ply1asclunit.4 $e |- .0. = ( 0g ` R ) $.
+    ply1asclunit.5 $e |- ( ph -> R e. Field ) $.
     ${
-      ressply1sub.1 $e |- ( ph -> X e. B ) $.
-      ressply1sub.2 $e |- ( ph -> Y e. B ) $.
-      $( A restricted polynomial algebra has the same subtraction operation.
-         (Contributed by Thierry Arnoux, 30-Jan-2025.) $)
-      ressply1sub $p |- ( ph -> ( X ( -g ` U ) Y ) = ( X ( -g ` P ) Y ) ) $=
-        ( cfv wcel cminusg cplusg co csg ressply1invg oveq2d wa wceq cgrp csubg
-        cbs csubrg subrgply1 subrgsubg subggrp syl ressply1bas eleqtrd grpinvcl
-        3syl eqid syl2anc eleqtrrd ressply1add mpdan eqtrd grpsubval 3eqtr4d
-        jca ) AIJGUASZSZGUBSZUCZIJCUASZSZCUBSZUCZIJGUDSZUCZIJCUDSZUCZAVMIVOVLUC
-        ZVQAVKVOIVLABCDEFGHJKLMNOPRUEUFAIBTZVOBTZUGWBVQUHAWCWDQAVOCUKSZBACUITZJ
-        WETZVOWETABEUJSTZWFAFDULSTBEULSTWHOBDEFGHKLMNUMBEUNUTBECPUOUPAJBWERABCD
-        EFGHKLMNOPUQZURZWECVNJWEVAZVNVAZUSVBWIVCVIABCDEFGHIVOKLMNOPVDVEVFAWCJBT
-        VSVMUHQRBVLGVJVRIJNVLVAVJVAVRVAVGVBAIWETWGWAVQUHAIBWEQWIURWJWEVPCVNVTIJ
-        WKVPVAWLVTVAVGVBVH $.
-    $}
+      ply1asclunit.6 $e |- ( ph -> Y e. B ) $.
+      ply1asclunit.7 $e |- ( ph -> Y =/= .0. ) $.
+      $( A non-zero scalar polynomial is a unit.  (Contributed by Thierry
```

Comment:
"A non-zero scalar polynomial **over a field** ..."

Reply:
To choose `R` as symbol for a field is a little bit misleading (usually used for a "ring").

Reply:
Actually, if @icecream17 and @benjub manage to remove the commutativity requirement in e.g. [psrass23](https://us.metamath.org/mpeuni/psrass23.html) by #4658, a division ring would be sufficient, since it would also trickle down to [ply1assa](https://us.metamath.org/mpeuni/ply1assa.html)....

Reply:
Fixed in https://github.com/metamath/set.mm/pull/4788/commits/e258c749150df69ed095ab2de998ecef5b2ab046 by adding the mention "over a field... ".

I thought about renaming the variable, but then there is the hope that this theorem is later reworked for division rings, so I'd propose to keep it like this for now.

Reply:
I don't see how this is possible. 
```
(𝑋 × (𝐴 · 𝑌)) = (𝐴 · (𝑋 × 𝑌))
```

If all of them are constant polynomials, then · = × and therefore the scalar ring is commutative